### PR TITLE
add falcon thread view

### PR DESCRIPTION
**Pull request by danilito19**
This new view will be used to assert threading functionality works as expected. Companion private PR will be shared.

### CODE REVIEW

File: apps/falcon_app.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 from hashlib import md5
 import os
-
+import threading
+import sys
 import falcon
 import vulnpy.falcon
-
+import time
+import contrast
```
Comment (lazorchakp) [inline apps/falcon_app.py:9]: these apps are intended to be run with or without contrast, so I don't think we should have it imported at the top level

**Review by lazorchakp — CHANGES_REQUESTED** [review]
Is there any reason to add this to `vulnpy` rather than to the falcon apptest app directly? It feels a little too agent-specific for this repo to me

**Comment by danilito19** [comment]
> Is there any reason to add this to `vulnpy` rather than to the falcon apptest app directly? It feels a little too agent-specific for this repo to me

because it's for integration tests ... I agree, if you have a better suggestion I"m all ears

**Comment by lazorchakp** [comment]
@danilito19 🤦  that makes sense. There are definitely ways around this but they're far beyond the scope of this work

File: apps/falcon_app.py [hunk 2]
```diff
@@ -31,10 +32,49 @@ def on_post(self, req, resp):
...
+    """This is intented to test Contrast specific functionality!"""
+
+    print("starting background thread")
+
+    # We're testing that a child thread still has a request context
+    # even after the parent thread exited, so waiting here so the parent
+    # thread has time to exit.
+    time.sleep(2)
+
+    import contrast
```
Comment (drdavella) [inline apps/falcon_app.py:45]: Probably no one else will use this view, but do we still need to handle `ImportError` here?

Comment (danilito19) [reply apps/falcon_app.py:45]: Yeah I can do that so it doesn't break.

**Review by drdavella — APPROVED** [review]

**Review by lazorchakp — APPROVED** [review]

**Pull request merged**